src/lib.rs

```diff
@@ -430,8 +479,13 @@ impl<K: Hash + Eq, V, S: HashState> LinkedHashMap<K, V, S> {
     /// assert_eq!(None, iter.next());
     /// ```
     pub fn iter(&self) -> Iter<K, V> {
+        let head = if self.head != ptr::null_mut() {
+            unsafe { (*self.head).prev }
+        } else {
+            ptr::null_mut()
```

Comment:
If the struct hasn't allocated the guard node yet we may dereference a null pointer at line 483